Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py

```diff
@@ -932,25 +932,34 @@ def isolate_endpoint_command(client, args):
         raise ValueError(f'Endpoint {endpoint_id} was not found')
 
     endpoint = endpoint[0]
+    endpoint_status = endpoint.get('endpoint_status')
     is_isolated = endpoint.get('is_isolated')
     if is_isolated == 'AGENT_ISOLATED':
         return (
-            f'Endpoint {endpoint_id} already isolated',
+            f'Endpoint {endpoint_id} already isolated.',
             None,
             None
         )
-
     if is_isolated == 'AGENT_PENDING_ISOLATION':
         return (
-            f'Endpoint {endpoint_id} pending isolation',
+            f'Endpoint {endpoint_id} pending isolation.',
             None,
             None
         )
-
+    if endpoint_status == 'DISCONNECTED':
+        return_error(
+            f'Error: Endpoint {endpoint_id} is disconnected and therefore can not be isolated.'
+        )
+    if is_isolated == 'AGENT_PENDING_ISOLATION_CANCELLATION':
+        return_error(
```

Comment:
instead of `return_error` raise `ValueError`

Reply:
there is a main try/except that will catch this exception and return_error.
The main reason why we prefer not to use return_error everywhere in the code because it has side effects - it uses sys.exit()

Reply:
done

Reply:
I still dont see that code has changed and that you `raise ValueError`

Reply:
im working on the unittest, cause its different with raise ValueError